eventio/iact/__init__.py

```diff
@@ -2,49 +2,78 @@
 import logging
 import numpy as np
 from collections import namedtuple
-
-from ..base import known_objects, EventIOFile
 from ..exceptions import WrongTypeException
-from .objects import (
-    CorsikaRunHeader,
-    CorsikaTelescopeDefinition,
-    CorsikaEventHeader,
-    CorsikaArrayOffsets,
-    CorsikaTelescopeData,
-    IACTPhotons,
-    IACTLayout,
-    IACTTriggerTime,
-    IACTPhotoElectrons,
-    CorsikaEventEndBlock,
-    CorsikaRunEndBlock,
-    CorsikaLongitudinal,
-    CorsikaInputCard,
-)
+log = logging.getLogger(__name__)
+from .objects import parse_eventio_object
+from ..event_io_file import objects
+
+def sort_objects_into_showers(objects):
+    '''
+    The objects in IACT run files build up showers 
+    when looking at the object.headers[-1].type of a list of objects
+    (where the first 3 and the last 1 object was stripped off already)
+    we find a shower looks like this: 
+    [...
+     1202,
+     1203,
+     1205,
+     ...
+     1205,
+     1209,
+     ...
+    ]
+    '''
+    showers = []
+    while True:
+        try:
+            idx = [o.headers[-1].type for o in objects].index(1209) + 1
+        except:
+            break
+        new_shower, objects = objects[:idx], objects[idx:]
+        showers.append(new_shower)
+    return showers
+
+
+def generate_event(shower):
+    '''
+    A shower is simply a list of EventIOObject
+    of these kinds:
```

Comment:
I will try to create one. If I understood the manual correctly, PhotoElectron bunches are written if the `CEFFIC` option is used.